### PR TITLE
Rename plugin ID to grafana-testdata-datasource

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Grafana.com does not allow plugins to not follow the convention of `org-plugin-type` for plugin IDs so this PR changes the plugin ID and add the hardcoded `Alias` so the current data sources and dashboards keep working.

**Why do we need this feature?**

It's not possible to publish the plugin standalone otherwise.

**Which issue(s) does this PR fix?**:

Ref https://github.com/grafana/grafana/issues/68973

### CODE REVIEW

File: .golangci.toml [hunk 1]
```diff
@@ -27,7 +27,10 @@ deny = [
 deny = [
   { pkg = "github.com/grafana/grafana/pkg/", desc = "Core plugins are not allowed to depend on Grafana core packages" },
 ]
-files = ["**/public/plugins/testdata/*", "**/public/plugins/testdata/**/*"]
+files = [
+  "**/public/plugins/grafana-testdata-datasource/*",
```
Comment (marefr) [inline .golangci.toml:31]: Nit. Would a more generic glob work here, like `**/public/plugins/*/**`?

File: .eslintrc [hunk 2]
```diff
@@ -91,7 +91,10 @@
       }
     },
     {
-      "files": ["public/plugins/testdata/*.{ts,tsx}", "public/plugins/testdata/**/*.{ts,tsx}"],
+      "files": [
```
Comment (marefr) [inline .eslintrc:94]: Would a more generic glob pattern work here to catch any/all plugins?

**Review by marefr — APPROVED** [review]
LGTM

**Review by ryantxu — APPROVED** [review]
🎉

**Pull request merged**